File: servers/dotnet/Domain/Blogs/Entities/Blog.cs
using Domain.Users.Entities;

namespace Domain.Blogs.Entities;

public class Blog
{
    public required string Id { get; set; }
    public required string Title { get; set; }
    public required string Body { get; set; }
    public DateTime CreatedAt { get; set; }

    public required string UserId { get; set; }
    public User User { get; set; } = null!;

    public static Blog NewInstance(string title, string body, string userId)
    {
        return new Blog
        {
            Id = Guid.NewGuid().ToString(),
            Title = title,
            Body = body,
            UserId = userId,
            CreatedAt = DateTime.UtcNow
        };
    }

}

File: servers/springboot/server/src/main/java/com/gabrielgavrilov/ohmyblog/core/blog/entity/Blog.java
package com.gabrielgavrilov.ohmyblog.core.blog.entity;

import com.gabrielgavrilov.ohmyblog.core.user.entity.User;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "blog")
@Getter
@Setter
@Accessors(chain = true)
public class Blog {

    @Id
    @Column(name = "blog_id")
    @Setter(AccessLevel.NONE)
    private UUID blogId;

    @Column(name = "user_id")
    @Setter(AccessLevel.NONE)
    private UUID userId;

    @Column(name = "title")
    private String title;

    @Column(name = "body")
    private String body;

    @Column(name = "created_at")
    private Instant createdAt;

    @ManyToOne()
    @Setter(AccessLevel.NONE)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    public static Blog newInstance(UUID userId, String title, String body) {
        Blog blog = new Blog();
        blog.blogId = UUID.randomUUID();
        blog.userId = userId;
        blog.title = title;
        blog.body = body;
        blog.createdAt = Instant.now();
        return blog;
    }

}
